### ISSUE TITLE
[Bug]: Delete property from Optional-Chaining 

### ISSUE THREAD

**Issue by OlegLustenko** [microsoft/TypeScript#35084]
**TypeScript Version:**  3.8.0-dev.20191112, 3.7.2

**Search Terms:** 
- https://github.com/microsoft/TypeScript/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+label%3ABug+delete
- https://github.com/microsoft/TypeScript/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+label%3ABug+optional-chaining

**Code**
```ts
interface z {
    name?: string;
}
var z: z = {
    name: 'qwe'
}
delete z?.name;
console.log(z);
```

**Expected behavior:** property is deleted from the object
**Actual behavior:** property still exists in the object
**Playground Link:** http://www.typescriptlang.org/play/?ts=3.8.0-dev.20191112#

**Related Issues:** I haven't found any related bugs

compiles to:
```ts
...
delete ((_a = z) === null || _a === void 0 ? void 0 : _a.name);
```
I think it should compile to something like this:
```ts
((_a = z) === null || _a === void 0 ? void 0 : delete _a.name);
```

**Comment by OlegLustenko** [comment]
**Chrome Canary**: Version 80.0.3966.1 (Official Build) canary-dcheck (32-bit)

![image](https://user-images.githubusercontent.com/11524061/68761710-50583e00-061d-11ea-832a-bfcab837184b.png)

**Comment by nmain** [comment]
Definitely a bug; this is called out as something that's specifically supposed to work in the [tc39 proposal](https://github.com/tc39/proposal-optional-chaining#optional-deletion)

**Comment by ajafff** [comment]
fix is up at #35090
